### ISSUE TITLE
Why not resume() ?

### ISSUE THREAD

**Issue by gvilarino** [indutny/sticky-session#31]
According to [the official docs](https://nodejs.org/api/net.html#net_net_createserver_options_connectionlistener), when `pauseOnConnect` is `true`, incoming socket connections are paused and need to be `resume()`'d in order for data to be read from them. `api.js` doesn't seem to be handling that in that way, and i can't seem to find the call to `resume()` elsewhere in the codebase. Is this on purpose?

**Comment by indutny** [comment]
@gvilarino they are resumed automatically when they are received by workers. This is on purpose! Thanks for asking.

**Comment by gvilarino** [comment]
Thanks for clarifying :D

About to try this on prod :fingers_crossed:

**Comment by indutny** [comment]
Mine are too!

**Comment by elad** [comment]
@indutny that's interesting to know! I wonder if that might lead to https://github.com/nodejs/node-v0.x-archive/issues/25594 where some connections still fail. I'm not as familiar with node.js internals, but maybe sending data before the process explicitly signaled it's ready for it might be a problem?

**Comment by indutny** [comment]
@elad this is very interesting. The thing is that it shouldn't really read any data from socket when the server was configured with `pauseOnConnect: true`. Pausing happens not on JS level, but rather on C++ level.

**Comment by elad** [comment]
@indutny I understand. My reasoning was based on the original problem leading to the introduction of `pauseOnConnect`: a connection was `accept()`ed and data started flowing, leading to a race condition between when the data was coming in and when the child process received the file descriptor and could read it. If it's `resume()`d automatically then maybe the window is shorter (the child process already has the socket) but still there.

**Comment by indutny** [comment]
It is resumed automatically in case of the regular http.Server without cluster at all. I don't think that there is any reason to think that it might be faulty.

The process is following:

1. Accept socket
2. ReadStart (resume) socket
3. `emit('connection')`

The data will be read from the socket only on a next libuv tick, and it will be queued into the javascript Duplex stream instance.

TL;DR: don't be worried ;)

**Comment by elad** [comment]
Okay, thanks. :)

**Comment by gvilarino** [comment]
thanks for clarifying all this stuff. I've learnt a lot today :raised_hands:
